text/0000-export-to-oci.md

```diff
@@ -0,0 +1,332 @@
+
+Let's see some examples of the proposed behavior
+
+## Examples
+
+For each case, I will present two ways of invoking the new feature:
+
+- Using the environment Variable
+- Using the new flag
+
+For both ways the expected output is the same
+
+### Analyzing run-image from OCI layout format
+
+Let's suppose the `<oci-dir>` or `CNB_OCI_PATH` environment variable points to a  directory that contains the `run-image` with the following format
+
+```=shell
+oci
+├── cnbs
+   └── sample-stack-run
+       ├── blobs
+       │   └── sha256
+       │       ├── 1f59171fcf9a8c2a78192d4dfbf88e6f0258e24f21798ffe015c07682d3a944a
+       │       ├── 219480c73c20efd82d425207e7555eba09cb113c845250cef09ba376e1dd506e
+       │       ├── 63882e64890d1adea5824dbb9bc6a812140d6b85752bf74a0a26fca78e1baf5a
+       │       ├── 63cb781c9d22ed765584437fea6db568f79be248ce40fc6695017d3ef3e8caaa
+       │       └── 6b534265b45a4c09a5e05dde5ccf2450ea505481658a1c9ae9e0aae32362e941
+       ├── index.json
+       └── oci-layout
+```
+
+And the analyzer is invoked as follows
+
+```=shell
+> export CNB_USE_OCI=true
+> /cnb/lifecycle/analyzer -run-image cnbs/sample-stack-run:bionic my-app-image
+```
+
+Or
+
+```=shell
+> /cnb/lifecycle/analyzer -oci -run-image cnbs/sample-stack-run:bionic my-app-image
+```
+
+The expected output is the analysis metadata [analyzed.toml](https://github.com/buildpacks/spec/blob/main/platform.md#analyzedtoml-toml) with a content similar to this one:
+
+```=TOML
+[run-image]
+  reference = "sha256:eed6d69be53111ad1d7d3f5d1c037350e6807986feb479a67f36b15f9205a56d"
+```
+
+Where the [ImageID](https://github.com/opencontainers/image-spec/blob/main/config.md#imageid) is calculated according to the OCI specification
+
+
+### Exporting to OCI layout format
+
+Similar to the previous example, let's suppose the `<oci-dir>` or `CNB_OCI_PATH` environment variable points to a  directory that contains the `run-image` with the following format
+
+```=shell
+oci
+├── cnbs
+   └── sample-stack-run
+       ├── blobs
+       │   └── sha256
+       │       ├── 1f59171fcf9a8c2a78192d4dfbf88e6f0258e24f21798ffe015c07682d3a944a
+       │       ├── 219480c73c20efd82d425207e7555eba09cb113c845250cef09ba376e1dd506e
+       │       ├── 63882e64890d1adea5824dbb9bc6a812140d6b85752bf74a0a26fca78e1baf5a
+       │       ├── 63cb781c9d22ed765584437fea6db568f79be248ce40fc6695017d3ef3e8caaa
+       │       └── 6b534265b45a4c09a5e05dde5ccf2450ea505481658a1c9ae9e0aae32362e941
+       ├── index.json
+       └── oci-layout
+```
+
+And the exporter is invoked as follows
+
+```=shell
+> export CNB_OCI_PATH=true
+> /cnb/lifecycle/exporter -run-image cnbs/sample-stack-run:bionic my-app-image
+```
+
+Or
+
+```=shell
+>  /cnb/lifecycle/exporter -oci -run-image cnbs/sample-stack-run:bionic my-app-image
```

Comment:
I've been thinking on this, I have the following idea: All the conversion logic should be handled by the platform right? because they have to **prepare** the inputs for lifecycle. Now, from the Lifecycle side, I like the idea to consider *asking for paths only*, based on that,  I am wondering the following:

Checking the **creator** phase (because that one has all the parameters required by the other phases), these are the current arguments (relevant for this RFC):

```=bash
 -cache-dir string
    	path to cache directory
  -cache-image string
    	cache image tag name
  -previous-image string
    	reference to previous image
  -run-image string
    	reference to run image
```
Also the **image name** it's important as you mentioned. We can add the following new arguments, valid only if the flag `oci` is enabled:

```=bash
  -previous-image-dir string
    	path to the previous image in OCI layout format. Relative to <oci-dir>
  -run-image-dir string
    	path to the run image in OCI layout format. Relative to <oci-dir>
  -image-dir string
       path to save the image in OCI layout format. Relative to <oci-dir>
```
In this way, we can use the actual **image name** as reference to update `org.opencontainers.image.ref.name`